src/template_partials/templatetags/partials.py

```diff
@@ -153,7 +173,13 @@ def partial_func(parser, token):
         tag_name, partial_name = token.split_contents()
     except ValueError:
         raise template.TemplateSyntaxError(
-            "%r tag requires a single argument" % token.contents.split()[0]
+            "%r tag requires a single argument" % tag_name
         )
 
-    return RenderPartialNode(partial_name, origin=parser.origin)
+    try:
+        extra_data = getattr(parser, "extra_data")
+        partial_mapping = SubDictionaryWrapper(extra_data, "template-partials")
+    except AttributeError:
+        partial_mapping = parser.origin.partial_contents
```

Comment:
We've pulled the access of `.partial_contents` up to parse time. It needs to be deferred to render-time too.

Reply:
I think we are not using `SubDictionaryWrapper` for Django before Django 5.1. We are using just partial_contents, which may be the issue?

Reply:
Yes, exactly. 

At the moment we're trying to access this it's not been defined (in the case where the partial is used before it's defined in the template)

Previously we passed the `origin` to the node, and only accessed the partial contents mapping at render time (by which time the template is fully parsed)

Reply:
I tried to implement it. Tests are passing now.